### PR TITLE
scheduler use individual thread handle high priority commands

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**

### CODE REVIEW

File: src/storage/mod.rs [hunk 1]
```diff
@@ -232,6 +232,26 @@ impl Command {
         }
     }
 
+    pub fn priority(&self) -> CommandPri {
+        match *self {
+            Command::Get { ref ctx, .. } |
```
Comment (andelf) [inline src/storage/mod.rs]: better add get_ctx() member function to `Command`

File: src/storage/mod.rs [hunk 2]
```diff
@@ -232,6 +232,14 @@ impl Command {
         }
     }
 
+    pub fn priority(&self) -> CommandPri {
+        self.get_context().get_priority()
+    }
+
+    pub fn need_flow_control(&self) -> bool {
+        !self.readonly() && self.priority() != CommandPri::High
```
Comment (siddontang) [inline src/storage/mod.rs]: if we have too many high priority commands, maybe we also need flow control.

Comment (zhangjinpeng1987) [inline src/storage/mod.rs]: There are few high priority commands, like system commands.

Comment (siddontang) [inline src/storage/mod.rs]: if the client has a bug, sending all command with high priority, how can we know this?

Comment (siddontang) [inline src/storage/mod.rs]: Do we have any a counter to count different priority commands?

File: src/storage/mod.rs [hunk 3]
```diff
@@ -1026,4 +1034,46 @@ mod tests {
         rx.recv().unwrap();
         storage.stop().unwrap();
     }
+
+    #[test]
+    fn test_high_priority_get_put() {
```
Comment (siddontang) [inline src/storage/mod.rs]: any test to check normal commands can't block high command?

File: src/storage/mod.rs [hunk 4]
```diff
@@ -206,6 +207,9 @@ impl Display for Command {
             Command::RawGet { ref ctx, ref key } => {
                 write!(f, "kv::command::rawget {:?} | {:?}", key, ctx)
             }
+            Command::Pause { ref ctx, duration } => {
```
Comment (siddontang) [inline src/storage/mod.rs]: can we use Pause in test only? 

/cc @BusyJay

Comment (zhangjinpeng1987) [inline src/storage/mod.rs]: @BusyJay any suggestions?

Comment (BusyJay) [inline src/storage/mod.rs]: You can put a `#[cfg(test)]` to every place `Pause` shows up. But I recommend to use trait instead when we are about to refactor it with future.

Comment (zhangjinpeng1987) [inline src/storage/mod.rs]: As I and jay have discussed, we will use future and no need add ugly `#[cfg(test)]`.

**Comment by siddontang** [comment]
LGTM

PTAL @BusyJay @disksing

File: src/storage/metrics.rs [hunk 5]
```diff
@@ -64,6 +64,13 @@ lazy_static! {
             &["type"]
         ).unwrap();
 
+    pub static ref SCHED_COMMANDS_PRI_COUNTER_VEC: CounterVec =
+        register_counter_vec!(
+            "tikv_scheduler_commands_pri_total",
+            "Total count of difference priority commands",
```
Comment (disksing) [inline src/storage/metrics.rs]: difference -> different.

**Comment by disksing** [comment]
LGTM.

**Pull request merged**